### PR TITLE
add a nan protection to avoid crash in fastjet call from pvsorter

### PR DESCRIPTION
**Pull request by arizzi**
this should address 
https://hypernews.cern.ch/HyperNews/CMS/get/recoDevelopment/1451/2/1/1/1.html
by adding a protection on "nan" being passed to fastjet.

the nan seems to be coming from ptError and should be further investigated

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@arizzi I'm still expecting 81X PR
Please clarify when or if.

File: CommonTools/RecoAlgos/src/PrimaryVertexSorting.cc [hunk 1]
```diff
@@ -27,7 +27,13 @@ float PrimaryVertexSorting::score(const reco::Vertex & pv,const  std::vector<con
     float scale=1.;
     if(c->bestTrack() != 0)
       {
-        scale=(c->pt()-c->bestTrack()->ptError())/c->pt();
+	if(c->pt()!=0) {
+       		 scale=(c->pt()-c->bestTrack()->ptError())/c->pt();
+	}
+ 	if(isnan(scale)) {
```
Comment (cvuosalo) [inline CommonTools/RecoAlgos/src/PrimaryVertexSorting.cc]: The static analyzer warns: "std::isnan does not work when fast-math is used. Please use edm::isNotFinite from 'FWCore/Utilities/interface/isNotFinite.h'".
Please fix this line.

**Comment by cvuosalo** [comment]
@arizzi: When might this PR get updated?

**Comment by cvuosalo** [comment]
@arizzi: When will the 81X version of this PR be submitted?

**Comment by arizzi** [comment]
81X PR made (just cherry picking the two commits, no other test performed)

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
+1

For #14846 365c9764ebbe78aaa79386f34f605242d7f7b88f

Bug fix to prevent crash due to a NAN in FastJet call from PV sorter. There should be no change in monitored quantities.

#14945 is the 81X version of this PR.

The code fix is satisfactory, and Jenkins tests against baseline CMSSW_8_0_X_2016-06-20-1100 show no significant differences, as expected.

**Comment by davidlange6** [comment]
+1

**Pull request merged**